src/App.tsx

```diff
@@ -1,24 +1,283 @@
+  }, [errorMessage]);
+
+  const handleCheckboxChange = (todoId: number) => {
+    const updatedTodos = todos.map(todo => {
+      if (todo.id === todoId) {
+        return { ...todo, completed: !todo.completed };
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  const handleAddTodo = useCallback(async (title: string) => {
+    setIsLoading(true);
+
+    try {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await todoService.createTodo(newTodo);
+
+      setTodos((currentTodos) => [...currentTodos, createdTodo]);
+    } catch {
+      setErrorMessage(ErrorMessage.ADD);
+    } finally {
+      setTempTodo(null);
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleRemoveTodo = useCallback(async (todoId: number) => {
+    setIsLoading(true);
+
+    try {
+      setTempTodo({
+        id: todoId,
+        title: '',
+        userId: USER_ID,
+        completed: false,
+      });
+
+      await todoService.deleteTodo(todoId);
+
+      setTodos((currentTodos) => currentTodos.filter((todo) => todo.id !== todoId));
+    } catch {
+      setErrorMessage(ErrorMessage.DELETE);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleClearAllCompletedTodos = useCallback(async () => {
+    todos
+      .filter((todo) => todo.completed)
+      .map((todo) => handleRemoveTodo(todo.id));
+  }, [todos]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setErrorMessage(ErrorMessage.TITLE);
+    }
+
+    handleAddTodo(todoTitle);
+    setTodoTitle('');
+  };
+
+  const makeSetFilterStatus = useCallback(
+    (filter: Status) => () => {
+      setFilterStatus(filter);
+    },
+    [],
+  );
+
+  const [completedTodos, uncompletedTodos] = useMemo(() => {
+    const completed = todos?.filter((todo) => todo.completed);
+    const uncompleted = todos?.filter((todo) => !todo.completed);
+
+    return [completed, uncompleted];
+  }, [todos]);
+
+  const todoIsActive = todos?.find((todo) => todo.completed === false);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterStatus) {
+      case Status.COMPLETED:
+        return completedTodos;
+
+      case Status.ACTIVE:
+        return uncompletedTodos;
+
+      case Status.ALL:
+      default:
+        return todos;
+    }
+  }, [todos, filterStatus]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
```

Comment:
The App component seems too big. Can you divide it into the smallest ones? TodoList, Footer, etc